src/App.tsx

```diff
@@ -1,26 +1,27 @@
+import { useEffect } from 'react';
 import { useParams } from 'react-router-dom';
 
-import Header from './containers/Header';
 import Loading from './containers/Loading';
 import NotFound from './containers/NotFound';
-import Successful from './containers/Successful';
 import PaymentGatewayMultiStep from './containers/PaymentGatewayMultiStep';
 
+import { useAppSelector } from './hooks/useRedux';
+import getOrderData from './hooks/useOrderData';
+
 import gradient from '/images/gradient.png';
-import useGetOrderData from './utils/getOrderData';
-import calculateRemainingTime from './utils/calculateRemainingTime';
 
 export default function App() {
   const { id } = useParams();
 
-  const { loading, data, error } = useGetOrderData('66dced5d0a1e1ccd057c0a1c');
-  const isExpired = !!calculateRemainingTime(data?.expiredTimestamp as number);
+  useEffect(() => {
+    getOrderData(id);
+  }, [id]);
+
+  const { loading, data, error } = useAppSelector((state) => state.data);
 
   if (loading) {
     return <Loading />;
   }
-  const isSuccessful = false;
-  // Todo implement is Successful in back so if the order is successful we know here
 
```

Comment:
add a line here and say `if (!id) { return <NotFound /> }`

and fix all the `id || ''` after that.